### ISSUE TITLE
WP_JSON_Comments_Controller::prepare_item_for_update() is too restrictive

### ISSUE THREAD

**Issue by danielbachhuber** [WP-API/WP-API#940]
All fields should be updateable, and `WP_JSON_Comments_Controller` should have one method for preparing data for the database.

**Comment by rachelbaker** [comment]
@danielbachhuber this is an issue with `wp_update_comment()` not handling all of the comment object properties.  How do you suggest we handle this?

**Comment by danielbachhuber** [comment]
> How do you suggest we handle this?

Probably need a corresponding core patch then.

**Comment by rmccue** [comment]
Which specific fields are we missing support for here?

**Comment by joehoyle** [comment]
I'm not sure I follow the description and title on this ticker. How is `prepare_item_for_update` "too restrictive"?

Is the issue that `WP_REST_Comments` needs to prepare all it's data, then once it's run `wp_update_comment` it needs to manually set some other fields?

**Comment by danielbachhuber** [comment]
Sorry. Seems like the original description was unclear. The intent of this issue is that I should be able to update _any_ comment fields, which core doesn't currently allow.

**Comment by rmccue** [comment]
Looks like `wp_update_comment` accepts updating:
```
'comment_post_ID', 'comment_content', 'comment_author', 'comment_author_email', 'comment_approved', 'comment_karma', 'comment_author_url', 'comment_date', 'comment_date_gmt', 'comment_type', 'comment_parent', 'user_id'
```

Do you mean the core of the API (i.e. *we* don't support updating all of these)? I'm still kinda unclear here.

**Comment by danielbachhuber** [comment]
> Do you mean the core of the API (i.e. we don't support updating all of these)?

Er, I mean we only support updating a subset: https://github.com/WP-API/WP-API/blob/develop/lib/endpoints/class-wp-rest-comments-controller.php#L669

It looks like we also have an odd `prepare_item_for_update()` method we should probably break down.

**Comment by rachelbaker** [comment]
I will pick this up.

**Comment by rachelbaker** [comment]
Started in #1245, and should make Beta 2

**Comment by rachelbaker** [comment]
Fixed in #1245
